modules/currency-l1/src/main/scala/org/tessellation/currency/l1/Main.scala

```diff
@@ -0,0 +1,177 @@
+package org.tessellation.currency.l1
+
+import java.util.UUID
+
+import cats.Applicative
+import cats.effect.IO
+import cats.effect.kernel.Resource
+import cats.syntax.semigroupk._
+
+import scala.concurrent.duration._
+
+import org.tessellation.BuildInfo
+import org.tessellation.currency.l1.cli.method.{Run, RunInitialValidator, RunValidator}
+import org.tessellation.currency.l1.domain.snapshot.programs.CurrencySnapshotProcessor
+import org.tessellation.currency.l1.modules.{CurrencyPrograms, CurrencyStorages}
+import org.tessellation.currency.schema.currency.{CurrencyBlock, CurrencySnapshot, CurrencyTransaction}
+import org.tessellation.currency.{CurrencyKryoRegistrationIdRange, currencyKryoRegistrar}
+import org.tessellation.dag.l1.http.p2p.P2PClient
+import org.tessellation.dag.l1.infrastructure.block.rumor.handler.blockRumorHandler
+import org.tessellation.dag.l1.modules._
+import org.tessellation.dag.l1.{DagL1KryoRegistrationIdRange, StateChannel, dagL1KryoRegistrar}
+import org.tessellation.ext.cats.effect.ResourceIO
+import org.tessellation.ext.kryo.{KryoRegistrationId, MapRegistrationId}
+import org.tessellation.schema.cluster.ClusterId
+import org.tessellation.schema.node.NodeState
+import org.tessellation.schema.node.NodeState.SessionStarted
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.app.{SDK, TessellationIOApp}
+import org.tessellation.sdk.infrastructure.gossip.{GossipDaemon, RumorHandlers}
+import org.tessellation.sdk.resources.MkHttpServer
+import org.tessellation.sdk.resources.MkHttpServer.ServerName
+import org.tessellation.sdk.{SdkOrSharedOrKernelRegistrationIdRange, sdkKryoRegistrar}
+
+import com.monovore.decline.Opts
+import eu.timepit.refined.boolean.Or
+import fs2.Stream
+
+object Main
+    extends TessellationIOApp[Run](
+      s"Currency-l1",
+      s"Currency L1 node",
+      ClusterId(UUID.fromString("517c3a05-9219-471b-a54c-21b7d72f4ae5")),
+      version = BuildInfo.version
+    ) {
+
+  val opts: Opts[Run] = cli.method.opts
+
+  type KryoRegistrationIdRange = CurrencyKryoRegistrationIdRange Or SdkOrSharedOrKernelRegistrationIdRange Or DagL1KryoRegistrationIdRange
+
+  val kryoRegistrar: Map[Class[_], KryoRegistrationId[KryoRegistrationIdRange]] =
+    currencyKryoRegistrar.union(sdkKryoRegistrar).union(dagL1KryoRegistrar)
```

Comment:
why `dagL1` is required in currency L1?

Reply:
Currently all the abstractions for L1 functionality are in dag module. Hence currencyL1 depends on dagL1. We can extract it to a separate module as one of the next steps and have both currencyL1 and dagL1 depend on this common logic in a common module. But for the purpose of easier review I decided to keep the abstractions in dagL1 module for now (these were merged in the previous PR) so it's not like moving implementation and abstracting it away at the same time consequently obscuring what was actually changing in the logic. As to why we need `dagL1KryoRegistrar` there are class registered there that are needed for the abstraction e.g. p2p consensus input (which is being singed).